src/app/pages/common/entity/entity-wizard/entity-wizard.component.ts

```diff
@@ -250,41 +250,107 @@ export class EntityWizardComponent implements OnInit {
-      if (selectedOption) {
-        result = selectedOption.label;
+  generateSummaryHtml(data: any, isRoot = true): string {
+    const className = isRoot ? '' : 'wizard-ul';
+    let result = `<ul class="${className}">`;
+    Object.keys(data).forEach((key) => {
+      const value = data[key];
+      if (value) {
+        result += '<li>';
+        if (Array.isArray(value)) {
+          result += `<div>${key}: ${value.length}</div>`;
+        } else if (typeof value === 'object') {
+          result += `<label>${key}:</label>`;
+          result += this.generateSummaryHtml(data[key], false);
+        } else {
+          result += `<div>${key}: ${value}</div>`;
+        }
       }
-    } else if (Array.isArray(formControl.value)) {
-      let arrayValueCount = 0;
-      formControl.value.forEach((item) => {
-        const isNotEmptyArray = new EntityUtils().filterArrayFunction(item);
-        if (isNotEmptyArray) {
-          arrayValueCount++;
+      result += '</li>';
+    });
+    result += '</ul>';
+    return result;
+  }
+
+  getSummaryValue(fieldConfig: FieldConfig, formControl: AbstractControl): any {
+    let result: any;
+    let value: any;
+
+    if (!formControl) {
+      return null;
+    }
+    let key = fieldConfig.placeholder;
+    if (!key) {
+      key = fieldConfig.name;
+    }
+
+    if (fieldConfig.type == 'dict') {
+      if (fieldConfig.subFields) {
+        fieldConfig.subFields.forEach((subFieldConfig: FieldConfig) => {
+          const subFormControl = formControl.get(subFieldConfig.name);
+          const subValue = this.getSummaryValue(subFieldConfig, subFormControl);
+          if (!value) {
+            value = subValue;
+          } else {
+            value = { ...value, ...subValue };
+          }
+        });
+      }
+    } else if (fieldConfig.type == 'list') {
+      fieldConfig.listFields.forEach((listFieldConfig: FieldConfig[], index: number) => {
+        const listFormGroup = (formControl as FormArray).at(index);
+        let listValue: any;
+        listFieldConfig.forEach((subListFieldConfig: FieldConfig) => {
+          const subListFormGroup = listFormGroup.get(subListFieldConfig.name);
+          const subValue = this.getSummaryValue(subListFieldConfig, subListFormGroup);
+          if (!listValue) {
+            listValue = subValue;
+          } else {
+            listValue = { ...listValue, ...subValue };
+          }
+        });
+        if (listValue) {
+          if (!value) {
+            value = [];
+          }
+          value.push(listValue);
         }
       });
-      result = arrayValueCount;
+    } else {
+      value = formControl.value;
+      if (fieldConfig.type === 'select') {
+        const selectedOption = fieldConfig.options.find((option) => option.value == formControl.value);
+        if (selectedOption) {
+          value = selectedOption.label;
+        }
+      }
+    }
+
+    if (value) {
+      result = { [key]: value };
```

Comment:
How about making a separate component to show summary, instead of forming html in a method?

Reply:
Fixed

Reply:
Not exactly what I had in mind.
I was thinking about making a component, whose solely responsibility would be showing summary. And then taking all the code here (`updateSummary`, `getSummaryValue`) and moving it to this new component.
Also would like to avoid forming html in components - `generateSummaryHtml`.
Just so that we can separate concerns a little bit here.